vespalib/src/vespa/vespalib/stllike/hashtable.h

```diff
@@ -103,26 +103,68 @@ class hash_node {
     enum {npos=-1u, invalid=-2u};
     hash_node() : _node(), _next(invalid) {}
     hash_node(const V & node, next_t next=npos)
-        : _node(node), _next(next) {}
+        : _node(),
+          _next(next)
+    {
+        new (_node) V(node);
+    }
     hash_node(V &&node, next_t next=npos)
-        : _node(std::move(node)), _next(next) {}
-    hash_node(hash_node &&) noexcept = default;
-    hash_node &operator=(hash_node &&) noexcept = default;
-    hash_node(const hash_node &) = default;             // These will not be created
-    hash_node &operator=(const hash_node &) = default;  // if V is non-copyable.
+        : _node(),
```

Comment:
I think we can skip 'node' double-initialization by not initializing 'node' here, not having to rely too heavily on compiler optimizations.

Reply:
Fixed